repository-data/webfiles/src/main/resources/site/freemarker/common/eventlist-main-eventlist.ftl

```diff
@@ -0,0 +1,61 @@
+<#ftl output_format="HTML">
+<#include "../include/imports.ftl">
+
+<@hst.setBundle basename="rb.component.latest-events"/>
+
+<#if pageable?? && pageable.items?has_content>
+<div>
+    <h3 class="list-title"><@fmt.message key="headers.title"/></h3>
+
+    <ol class="list list--reset cta-list">
+        <#list pageable.items as item>
+        <li class="has-edit-button">
+            <article class="cta">
+                <h2 class="cta__title">
+                    <a href="<@hst.link hippobean=item/>" title="${item.title}">${item.title}</a>
+                </h2>
+
+                <#assign minStartTimeStamp = 0/>
+                <#assign maxEndTimeStamp = 0/>
+                <#-- Gather the earliest start date and the latest end date for each event -->
+                <#list item.events as event>
+                    <#-- Store the earliest start date values -->
+                    <#assign startTimeStamp = event.startdatetime.time?long/>
+                    <#if minStartTimeStamp == 0 || startTimeStamp lt minStartTimeStamp>
+                        <#assign minStartTime = event.startdatetime.time/>
+                        <#assign minStartTimeStamp = startTimeStamp/>
+                    </#if>
+
+                    <#-- Store the latest end date values -->
+                    <#assign endTimeStamp = event.enddatetime.time?long/>
+                    <#if maxEndTimeStamp == 0 || endTimeStamp gt maxEndTimeStamp>
+                        <#assign maxEndTime = event.enddatetime.time/>
+                        <#assign maxEndTimeStamp = endTimeStamp/>
+                    </#if>
+                </#list>
+
+                <@fmt.formatDate value=minStartTime type="Date" pattern="yyyy-MM-dd" pattern="yyyy-MM-dd" var="comparableStartDate" timeZone="Europe/London" />
```

Comment:
This may not necessarily be a matter for this particular pull request but it seems there is a number of statements like this, with patterns and hardcoded timezone, duplicated among a number of templates.

It would probably be better to come up with a custom macro or two, imported from a common file so that, when a requirement comes to change date format globally, or make the time zone dynamically recovered from user request (rather than hardcoded), there would be just one place to change them.

Reply:
I absolutely agree @ziemsky, I think it really makes sense to use a macro for timeZone now that we have it widely implemented. We'll deal with it and will make a separate PR.